### PR TITLE
document discard policy

### PR DESCRIPTION
**Pull request by scottf**

### CODE REVIEW

**Review by ripienaar — APPROVED** [review]
LGTM

File: adr/ADR-8.md [hunk 1]
```diff
@@ -221,6 +221,9 @@ A bucket is a Stream with these properties:
  * Maximum value sizes can be capped using `max_msg_size`
  * Maximum number of keys cannot currently be limited
  * Overall bucket size can be limited using `max_bytes`
+ * Discard Policy is always set to `new`
```
[the remark this answers is not in the document]
Comment (ripienaar) [reply adr/ADR-8.md:224]: Don't think we can realistically support both worlds - with KV not being GA, I think we can state this and the clients can reject earlier servers

https://github.com/nats-io/nats.go/blob/8045a5037b0128b90a293579d9d66de6c848c5f2/kv.go#L269

[the remark this answers is not in the document]
Comment (kozlovic) [reply adr/ADR-8.md:224]: Ok, so I read that we should have all libraries reject pre v2.7.2 servers and always set the discard policy to new. After such change in the libraries, though, we will have support requests likely, even if we explain in the error that they need to upgrade to v2.7.2+ and also likely update their stream from discardOld to discardNew, out of band.

[the remark this answers is not in the document]
Comment (ripienaar) [reply adr/ADR-8.md:224]: natscli.next will have `kv upgrade STREAM` for that, but yeah, i dont think there's really a great path eitherway :(

**Review by kozlovic — APPROVED** [review]
LGTM

**Pull request merged**